src/AutoMapper.Collection/Mappers/EquivalentExpressionAddRemoveCollectionMapper.cs

```diff
@@ -62,30 +62,45 @@ public class EquivalentExpressionAddRemoveCollectionMapper : IConfigurationObjec
             return destination;
         }
 
-        private static readonly MethodInfo MapMethodInfo = typeof(EquivalentExpressionAddRemoveCollectionMapper).GetRuntimeMethods().First(_ => _.IsStatic);
+        private static readonly MethodInfo _mapMethodInfo = typeof(EquivalentExpressionAddRemoveCollectionMapper).GetRuntimeMethods().Single(x => x.IsStatic && x.Name == nameof(Map));
 
         public bool IsMatch(TypePair typePair)
         {
             return typePair.SourceType.IsEnumerableType()
-                   && typePair.DestinationType.IsCollectionType()
-                   && this.GetEquivalentExpression(TypeHelper.GetElementType(typePair.SourceType), TypeHelper.GetElementType(typePair.DestinationType)) != null;
+                   && typePair.DestinationType.IsCollectionType();
         }
 
         public Expression MapExpression(IConfigurationProvider configurationProvider, ProfileMap profileMap, IMemberMap memberMap,
             Expression sourceExpression, Expression destExpression, Expression contextExpression)
         {
+            IObjectMapper nextMapper = _collectionMapper;
```

Comment:
Can next mapper be cached into concurrency dictionary so that it doesn't have to re-check every time?  For performance if there is no equivalency.